src/test/java/com/puppycrawl/tools/checkstyle/ant/CheckstyleAntTaskTest.java

```diff
@@ -527,6 +528,34 @@ public final void testXmlOutput() throws IOException {
         }
     }
 
+    @Test
+    public final void testSarifOutput() throws IOException {
+        final CheckstyleAntTask antTask = getCheckstyleAntTask();
+        antTask.setFile(new File(getPath(VIOLATED_INPUT)));
+        antTask.setFailOnViolation(false);
+        final CheckstyleAntTask.Formatter formatter = new CheckstyleAntTask.Formatter();
+        final File outputFile = new File("target/log.sarif");
+        formatter.setTofile(outputFile);
+        final CheckstyleAntTask.FormatterType formatterType = new CheckstyleAntTask.FormatterType();
+        formatterType.setValue("sarif");
+        formatter.setType(formatterType);
+        antTask.addFormatter(formatter);
+        antTask.execute();
+
+        final List<String> expected = readWholeFile(
+                new File(getPath("ExpectedCheckstyleAntTaskSarifOutput.sarif")));
+        final List<String> actual = readWholeFile(outputFile);
+        for (int lineNumber = 0; lineNumber < expected.size(); lineNumber++) {
+            final String line = expected.get(lineNumber);
+            if (line.trim().startsWith("\"uri\"")) {
+                continue;
```

Comment:
Skipping is not good approach.
Can create sub string after `**` of expected line and do contains in actual

Reply:
Not sure how the behaviour would be in win. Do we have a generic path comparing method in this project or Do you have a suggestion on how this has to be handled ?

Reply:
I am open for suggestions. I I just pointing that just ignoring is not very good idea.
Some  partial validation would be good.

Reply:
Normalized the windows path before comparing.